test/e2e/agent/agent_updates_test.go

```diff
@@ -67,6 +67,52 @@ var _ = Describe("VM Agent behavior during updates", func() {
 			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
 			logrus.Info("Device updated to new image 🎉")
 		})
+		It("Should update to v4 with embedded application", Label("updates", "OCP-77667"), func() {
+			// Check the device status right after bootstrap
+			response := harness.GetDeviceWithStatusSystem(deviceId)
+			device := response.JSON200
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+			Expect(*device.Status.Summary.Info).To(Equal("Bootstrap complete"))
+			Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+			var newImageReference string
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("current image for %s is %s", deviceId, currentImage)
+				repo, _ := parseImageReference(currentImage)
+				newImageReference = repo + ":v4"
```

Comment:
You need to add the script changes to create :v4 image

Reply:
What builds :v4 ? we need the script changes to do that :)

Reply:
v4 has an embedded app that tries to start a podman container with infinite sleep but fails to do so, its the one recomended by @hexfusion for this test

Reply:
https://github.com/flightctl/flightctl/blob/main/test/scripts/agent-images/create_agent_images.sh#L8